examples/example_glfw_vulkan/main.cpp

```diff
@@ -68,33 +69,77 @@ static VKAPI_ATTR VkBool32 VKAPI_CALL debug_report(VkDebugReportFlagsEXT flags,
 }
 #endif // IMGUI_VULKAN_DEBUG_REPORT
 
-static void SetupVulkan(const char** extensions, uint32_t extensions_count)
+// Check if the enumration extension has to be enabled
+bool checkExtensions(VkInstance instance)
+{
+    uint32_t extensionCount = 0;
+    vkEnumerateInstanceExtensionProperties(nullptr, &extensionCount, nullptr);
+    std::vector<VkExtensionProperties> availableExtensions(extensionCount);
+    vkEnumerateInstanceExtensionProperties(nullptr, &extensionCount, availableExtensions.data());
+
+    bool enablePortabilitySubset = false;
+    bool portabilitySubsetEnabled = false;
+    bool enableGetPhysicalDeviceProperties2 = false;
+    bool getPhysicalDeviceProperties2Enabled = false;
+
+#if defined(__APPLE__)
+    const bool macOS = true;
+#else
+    const bool macOS = false;
+#endif
+
+    for (const auto &extension : availableExtensions)
+    {
+        if (strcmp(extension.extensionName, VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0)
```

Comment:
Are all these `*_NAME` and `*_BIT_KHR` names defined in all older Vulkan SDKs that we support, or do they need to be guarded by the preprocessor, conditionally defined by us, or use our own string literals and constants?

Reply:
I would have to look it up, if all `*_NAME` and `*_BIT_KHR` are in older versions. I only looked at the newest version. Thanks for pointing this out.